services/autoscaling/src/simcore_service_autoscaling/utils_aws.py

```diff
@@ -58,49 +123,40 @@ def compose_user_data(settings: AwsSettings) -> str:
     )
 
 
-def create_ec2_client(settings: AwsSettings):
-    return boto3.client(
-        "ec2",
-        aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
-        aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
-        region_name=settings.AWS_REGION_NAME,
-    )
-
-
-def start_instance_aws(
+def start_aws_instance(
     settings: AwsSettings,
-    ami_id: str,
     instance_type: str,
-    tag: str,
-    service_type: str,
+    tags: list[str],
 ):
-    user_data = compose_user_data(settings)
-
-    ec2 = boto3.resource("ec2", region_name="us-east-1")
-    instance = ec2.create_instances(
-        ImageId=ami_id,
-        KeyName=settings.AWS_KEY_NAME,
-        InstanceType=instance_type,
-        SecurityGroupIds=settings.AWS_SECURITY_GROUP_IDS,  # Have to be parametrized
-        MinCount=1,
-        MaxCount=1,
-        InstanceInitiatedShutdownBehavior="terminate",
-        SubnetId=settings.AWS_SUBNET_ID,  # Have to be parametrized
-        TagSpecifications=[
-            {"ResourceType": "instance", "Tags": [{"Key": "Name", "Value": tag}]}
-        ],
-        UserData=user_data,
-    )[0]
-
-    logger.debug(
-        "New instance launched for %s services. Estimated time to launch and join the cluster : 2mns",
-        service_type,
-    )
-    logger.debug("Pausing for 10mns before next check")
-
-    time.sleep(600)
-    logger.debug("Instance state: %s", instance.state)
-    logger.debug("Public dns: %s", instance.public_dns_name)
-    logger.debug("Instance id: %s", instance.id)
-
-    return instance
+    with log_context(
+        logger,
+        logging.DEBUG,
+        msg=f"launching AWS instance {instance_type} with {tags=}",
+    ), ec2_client(settings) as client:
+        user_data = _compose_user_data(settings)
+
+        instances = client.run_instances(
+            ImageId=settings.AWS_AMI_ID,
+            MinCount=1,
+            MaxCount=1,
```

Comment:
Maybe to parametrize in the futur

Reply:
yes for a next version (see PR description)